CT-e

robertorp opened the pull request:
O objetivo deste pull e implementar o CT-e.

Simplesmente estou desenvolvendo ele ainda o mesmo sera usado pela empresa nossa aqui. 

Metodologia de desenvolvimento: 2.00 é 3.00 (incluindo CTeOs)


Pre-Requisito para iniciar as tarefas - Refatorar Solução para ter alguns projetos que sirvam para todos os de mais.
(Irei postar aqui todos os pull request que forem feito)
Unificação de classes de Assinatura https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/pull/332 OK
Unificar assinatura de NF-e é MDF-e (Data prevista 13/01/2017) https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/pull/341 OK
Unficar CertificadoDigital (essa vai dar um pouco de trabalho por causa das ultimas mudanças nela, tem que pensar uma maneira para unificar) https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/pull/347 OK
Unificar classe FuncoesXml https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/pull/357 OK

- [X] Importar tudo oque for necessário do projeto fornecidor pelo @leandrossantos  Data Prevista 12/02/2017 - OK
- [X] Construindo AppTeste para OK
- [x] Construindo Classes de Geração de XML de Recepção - Aguardando AppTeste para poder testar a geração do xml. OK

- [X] Serviço de Recepção de OK
- [X] Web Service – CteRetRecepcao OK
- [x] Web Service - CteInutilizacao OK
- [X] Web Service – CteConsulta Protocolo OK
- [X] Web Service – CteStatusServico OK
- [X] Sistema de Registro de Eventos OK
- [X] Evento de Cancelamento OK
- [X] Evento Carta de Correção OK
- [X] Conferir classes OK
- [X] Verificar namespaces OK
- [X] Fazer testes geral OK
- [X] Adicionar URLs de Produção - OK
- CTEOS Não sera feito por hora, tenho que começar a agilizar o mdf-e 3.0 (em teoria funciona mas.. na pratica não sei ainda)

Obs: não vou alterar nem uma linha de código fonte do NF-e em circunstância alguma. 
Obs2: não irei me preocupar com impressão por hora. (Nos da empresa aqui usamos o ACBR para impressão de documentos, atende bem)
Obs3: irei adicionando mais tópicos aqui de acordo com o andamento do desenvolvimento.

Com varias opiniões podemos chegar a um resultado perfeito.

robertorp committed:
Pre-projeto ct-e

marcosgerene commented:
Bom dia, assim que concluído as classes do XML farei a impressão em FastReport retrato do CTe e dos eventos.

leandrossantos commented:
@robertorp ou @marcosgerene  chamei vocês no Skype, quero ver com vocês sobre a parte de CTe, acredito que tenho uma boa parte adiantada ou até tudo pronta, faltando apenas alguns detalhes.

marcosgerene commented:
Ainda estou a passos de tartaruga, muito serviço, mas sempre que sobra um tempo vou mexer com as impressões, acredito que o MDFe sai até o meio do mês, quero acabar com minhas dependências externas

adeniltonbs commented:
@robertorp Unificação de classes de Assinatura #332 liberada.

robertorp committed:
Merge pull request #1 from robertorp/master

merge nos dados do master com o branch ct-e

marcosgerene commented:
#357 ok

robertorp committed:
merge

robertorp committed:
Merge pull request #2 from robertorp/master

Merge do master com o ZeusCT-e

robertorp committed:
Criado projetos para ct-e

robertorp committed:
Adicionado algumas classes que o leandro forneceu

robertorp commented:
Boa hehe logo logo vai ta tudo disponível =)

robertorp committed:
merge

robertorp committed:
add consulta

robertorp committed:
adicionado algumas informacoes no utils

robertorp committed:
adicionado arquivos no ct-e wsdl

robertorp committed:
adicionado algumas classes, revisar

robertorp committed:
identificação (ide) todo ajustado para versão 3 é versão 2

robertorp committed:
compl ajustado para versão 2.00 é 3.00

robertorp committed:
Emitente para 2.00 é 3.00

robertorp committed:
Ajustado remetente para versão 2.00 é 3.00

robertorp committed:
Ajustado expedidor para 2.00 é 3.00

robertorp committed:
Ajustado receb 2.00 é 3.00

robertorp committed:
Ajustado dest para suportar versões 2.00 é 3.00

robertorp committed:
Ajustado vPrest

robertorp committed:
ajustes na imp

robertorp committed:
Ajustes para suportar versão 2.0 é versao 3.0

robertorp committed:
Alterado layout para suportar modelo (67) CT-e Ordem de Serviço (CT-e OS)

robertorp committed:
layout rodoviario conforme versão 2.0 , 3.0 é 3.0 OS

robertorp committed:
ajustado modal aereo

robertorp committed:
criado modal aquav

robertorp committed:
layout multimodal e ferrov adicionado

robertorp committed:
modais concluido

robertorp committed:
Excluido alguns arquivos do git

robertorp committed:
criado base de app teste

robertorp committed:
salvando configurações do app teste

robertorp committed:
Feito Carregamento de configurações no app teste

robertorp committed:
wsdl status cte criado

robertorp committed:
feito extenção para versão
adicionado certificado na configuracaoservico
criado wsdlfactory

robertorp committed:
urls para homologação adicionado,
adicionado schemas
Consulta Status Servico Feito, tanto para 2.00 é 3.00

robertorp committed:
removido dlls, exe etc..

robertorp committed:
Feito Consulta Protocolo

robertorp committed:
Feito Inutilização

robertorp committed:
Feito consulta por recibo

robertorp committed:
Carta correção é Cancelamento
Não foram testados ainda, pois o mesmo esper um objeto CT-e
O objeto CT-e não esta sendo serializado ainda

robertorp committed:
Ajustado objeto CT-e
Testado Cancelamento e Carta correção - funcionando.

robertorp committed:
Feito código inicial de envio do ct-e

robertorp committed:
visual studio 2017

robertorp committed:
preenchendo xml de ct-e

robertorp committed:
gerado xml 2.00
faltando apenas fazer validação de schema,
envio

robertorp committed:
feito geração de xml de ct-e 2.00 é validado contra schemas

robertorp committed:
Gerado CT-e 3.00 também
de acordo com a configuração do app teste ele gera o respectivo cte

robertorp committed:
Envio de lote de ct-e 2.00 é 3.00 concluido

robertorp committed:
ajustes para salvar todos retornos em xmls

robertorp committed:
Adicionado licença de uso

robertorp committed:
Ajustes de namcespaces

robertorp committed:
ajustes de retornos

robertorp commented:
@adeniltonbs  Acho que finalizei rsrs faz uma verificação ai

Logico muito provavelmente vai precisar de alguns ajustezinhos mas isso é o menor dos problemas.. o maior já esta pronto.

Aceitando esse merge.. vou me dedicar no MDF-e 3.0 pois o mesmo já esta querendo raspar na trave o prazo rsrs (mês 06 a versão 1 sai do ar)

UPDATE faltou uma coisa.. na classe URL HELPER adicionar os links de PRODUÇÃO da CT-e vou Adicionar no dia 16/03 anoite! OBRIGADO rsrs! terminando faço outro update aqui hehe

marcosgerene commented:
@adeniltonbs aceitando o PR em 15 dias acredito que consigo fazer a impressão em Fast Reports para o CTe (rodoviario) e os eventos evitando dependência externa para impressão dos documentos mais comuns

robertorp committed:
Adicionado urls de produção

robertorp committed:
Salvar Xml CTe individualmente quando autorizar um lote (se marcado para salvar no app teste)

robertorp commented:
@marcosgerene  vlw man! mal posso esperar pra sair essa impressão hehe
@adeniltonbs  So conferir =)

robertorp committed:
Feito compatbilidade com c# 5

robertorp committed:
Merge remote-tracking branch 'upstream/master' into zeus-ct-e

robertorp committed:
ajustes para c# 5 pos merge com master

robertorp committed:
ajustes
